src/image_intel.cpp

```diff
@@ -20,29 +20,73 @@
 constexpr size_t IMAGE_A_ADDR = 0x20080000;
 constexpr size_t IMAGE_B_ADDR = 0x22480000;
 
+enum class BootPart
+{
+    image_a,
+    image_b,
+    unknown
+};
+
 /**
- * @brief Get current boot address
+ * @brief Get current boot partition
  */
-static size_t getBootAddress()
+static BootPart getBootPart()
 {
+    BootPart activePart = BootPart::unknown;
     const std::string bootcmdPrefix = "bootcmd=bootm";
+    std::string bootpart;
 
+    try
+    {
+        bootpart = exec(FW_PRINTENV_CMD " -n bootside 2>/dev/null");
+    }
+    catch (...)
+    {}
+    if (bootpart == "a\n")
+    {
+        return BootPart::image_a;
+    }
+    if (bootpart == "b\n")
+    {
+        return BootPart::image_b;
+    }
     std::string bootcmd = exec(FW_PRINTENV_CMD " bootcmd 2>/dev/null");
     if (bootcmd.compare(0, bootcmdPrefix.length(), bootcmdPrefix) == 0)
     {
-        return std::stoul(bootcmd.substr(bootcmdPrefix.length()), nullptr, 16);
+        size_t address =
+            std::stoul(bootcmd.substr(bootcmdPrefix.length()), nullptr, 16);
+        switch (address)
+        {
+            case IMAGE_A_ADDR:
+                activePart = BootPart::image_a;
+                break;
+            case IMAGE_B_ADDR:
+                activePart = BootPart::image_b;
+                break;
+        }
     }
 
-    return 0;
+    return activePart;
 }
 
 /**
- * @brief Set boot address for the next boot.
+ * @brief Set boot partition for the next boot.
  */
-static void setBootAddress(size_t address)
+static void setBootPart(BootPart part)
 {
-    std::ignore =
-        exec(FW_SETENV_CMD " bootcmd bootm %08x 2>/dev/null", address);
+    try
+    {
+        std::ignore = exec(FW_PRINTENV_CMD " -n bootside 2>/dev/null");
```

Comment:
Yes. This `exec` implementation trows exception in case command return with non-zero status. So the code suppress exception in a case `bootside` variable not defined.

Reply:
Probably, It would be better to leave a comment here.
It isn't so obvious and will be easily forgotten.

Reply:
done